firecrown/ccl/systematics/cl.py

```diff
@@ -138,15 +138,30 @@ def apply(self, cosmo, params, source):
     def _gen_selection_function(self, source):
 
         def _selfunc(lnmass, a):
+            a = np.atleast_1d(a)
```

Comment:
So this could be, for example mass as a function of richness and redshift with some scatter and/or cut-off?

Reply:
This is a selection function. So it is 1 at a given mass and scale factor If you count t he cluster and zero if not.